### PR TITLE
Update offset determination for ipad

### PR DESCRIPTION
**Pull request by imurchie**
Handle tabs and the Smart App Banner. Works for iPhone and iPad.

### CODE REVIEW

File: lib/commands/web.js [hunk 1]
```diff
@@ -14,6 +15,50 @@ extensions.getElementHeightMemoized = _.memoize(async function (key, el) {
   return (await this.getNativeRect(el)).height;
 });
 
+extensions.getExtraTranslateWebCoordsOffset = async function () {
+  let offset = 0;
+
+  let iphone = true;
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: I'd rather rename it to _isIPhone_

File: lib/commands/web.js [hunk 2]
```diff
@@ -14,6 +15,50 @@ extensions.getElementHeightMemoized = _.memoize(async function (key, el) {
...
+    }
+
+    // when scrolling has happened, there is a tick more offset needed
+    offset -= 10;
+  } finally {
+    // return implicit wait to what it was
+    this.setImplicitWait(implicitWaitMs);
+  }
+
+  // extra offset necessary (where do these come from? they just work)
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: :)

File: lib/commands/web.js [hunk 3]
```diff
@@ -30,18 +75,12 @@ extensions.getExtraNativeWebTapOffset = async function () {
...
-      // reload button found, which means scrolling has not happened
-    } catch (err) {
-      // no reload button, which indicates scrolling has happened
-      try {
-        const el = await this.findNativeElementOrElements('accessibility id', 'URL', false);
-        offset -= await this.getElementHeightMemoized('URLBar', el);
-      } catch (ign) {
-        // no URL elements found, so continue
-      }
+      const el = await this.findNativeElementOrElements('xpath', `//XCUIElementTypeButton[@name='Close app download offer']/following-sibling::XCUIElementTypeOther[1]`, false);
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: is it somehow possible to avoid xpath usage?

File: lib/commands/web.js [hunk 4]
```diff
@@ -52,6 +91,19 @@ extensions.getExtraNativeWebTapOffset = async function () {
...
+  let atomsElement = this.useAtomsElement(el);
+  let {x, y} = await this.executeAtom('get_top_left_coordinates', [atomsElement]);
+  let {width, height} = await this.executeAtom('get_size', [atomsElement]);
+  x = x + (width / 2);
+  y = y + (height / 2);
+
+  this.curWebCoords = {x, y};
+  await this.clickWebCoords();
+  // make sure real tap actually has time to register
+  await B.delay(500);
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: is there a way not hardcode the delay?

File: lib/commands/web.js (hunk 3 above)
Comment (mykola-mokhnach) [reply lib/commands/web.js]: this locator may also fail if localized browser is used

File: lib/commands/web.js [hunk 5]
```diff
@@ -14,6 +15,50 @@ extensions.getElementHeightMemoized = _.memoize(async function (key, el) {
...
+    // no reload button, which indicates scrolling has happened
+    try {
+      const el = await this.findNativeElementOrElements('accessibility id', 'URL', false);
+      offset -= await this.getElementHeightMemoized('URLBar', el);
+    } catch (ign) {
+      // no URL elements found, so continue
+    }
+
+    // when scrolling has happened, there is a tick more offset needed
+    offset -= 10;
```
Comment (jlipps) [inline lib/commands/web.js]: rather than putting `10` in a few places, can we give it a meaningful constant name?

**Pull request merged**